### ISSUE TITLE
`match_cancelled` does not prevent game from starting if `game_launch` has been received

### ISSUE THREAD

**Issue by Askaholic** [FAForever/downlords-faf-client#2543]
**Describe the bug**
1. Server sends `game_launch`
2. Client downloads map, or starts generating map
3. Server sends `match_cancelled`
4. Client starts the game

**To Reproduce**
With these server changes PR: https://github.com/FAForever/server/pull/882 you can reproduce by:
1. Make sure that the map pool is set up to use map generator only (this will ensure that there is a significant lag between receiving `game_launch` and starting the exe.
1. Join 2 players into the queue
2. The host closes their game as soon as it opens
3. The guest should see the map being generated and get the `match_cancelled` notification but their game still starts.

**Log or error message**
This is my log file:
[client.log](https://github.com/FAForever/downlords-faf-client/files/7921545/client.log)

In this log file we launched 2 games. The first one was cancelled by the guest after both players games had opened, and the second one was cancelled by the host how it is described in the **To Reproduce** section.

**Expected behavior**
It would be nice if the client were able to not launch the game after getting a `match_cancelled` message even if it has already started the pre-launch sequence.

**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[![Issue hunt](https://github.com/BoostIO/issuehunt-materials/raw/master/v1/issuehunt-button-v1.svg?sanitize=true)](https://issuehunt.io/r/FAForever/downlords-faf-client)

**Comment by Sheikah45** [comment]
Did you attach a log file?

**Comment by Askaholic** [comment]
No, do you need one? I could redo the test with debug logging if you want

**Comment by Sheikah45** [comment]
Thanks will take a look
